indexer/reindex_bags_backlog.py

```diff
@@ -0,0 +1,191 @@
+"""
+This script will send every bag in the storage service to the bags indexer
+to be re-indexed in Elasticsearch.
+"""
+
+import json
+import uuid
+
+import boto3
+from elasticsearch import Elasticsearch
+from elasticsearch_dsl import Search
+
+
+def scan_table(dynamo_client, *, TableName, **kwargs):
+    paginator = dynamodb_client.get_paginator("scan")
+
+    for page in paginator.paginate(TableName=TableName, **kwargs):
+        yield from page["Items"]
+
+def create_client(service_name, role_arn):
+    def _assumed_role_session(role_arn):
+        sts_client = boto3.client('sts')
+
+        assumed_role_object=sts_client.assume_role(
+            RoleArn=role_arn,
+            RoleSessionName=uuid.uuid1().hex
+        )
+
+        credentials=assumed_role_object['Credentials']
+
+        session = boto3.Session(
+            aws_access_key_id=credentials['AccessKeyId'],
+            aws_secret_access_key=credentials['SecretAccessKey'],
+            aws_session_token=credentials['SessionToken']
+        )
+
+        return session
+
+    session = _assumed_role_session(role_arn)
+    return session.client(service_name)
+
+# The bag indexer only cares about space, externalIdentifier & version
+def fake_known_replicas_payload(space, externalIdentifier, version):
+    return {
+        'context': {
+            'ingestId': "8368b576-8206-4552-870c-005186f0264a",
+            'ingestType': {
+                "id": "create"
+            },
+            'storageSpace': space,
+            'ingestDate': "2069-11-30T03:09:31.986Z",
+            'externalIdentifier': externalIdentifier,
+        },
+        'version': version,
+        'knownReplicas': {
+            'location': {
+                'provider': {
+                    "type": "AmazonS3StorageProvider"
+                },
+                'prefix': {
+                    "namespace": "go3dmZxK",
+                    "path": "XrLuVFbL"
+                }
+            },
+            'replicas': []
+        }
+
+    }
+
+def publish_payload(sns_client, topic_arn, payload):
+    json_payload = json.dumps(payload)
+
+    return sns_client.publish(
+        TopicArn=topic_arn,
+        Message=json_payload
+    )
+
+def get_total_bags(dynamodb_client, table_name):
+    resp = dynamodb_client.describe_table(TableName=table_name)
+    return resp["Table"]["ItemCount"]
+
+def get_latest_bags(dynamodb_client, table_name):
+    total_bags = get_total_bags(dynamodb_client, table_name)
+
+    print(f"total_bags: {total_bags}")
+
+    bags = {}
+    seen_bags = 0
+
+    for item in scan_table(dynamodb_client, TableName=table_name):
+        dynamo_id = item['id']['S']
+        version = int(item['version']['N'])
+        stored_version = bags.get(dynamo_id, -1)
+
+        seen_bags = seen_bags + 1
+
+        if(seen_bags % 10 == 0 or seen_bags > (total_bags - 10)):
+            print(f"{seen_bags}/{total_bags}")
+
+        if(version > stored_version):
+            bags[dynamo_id] = version
```

Comment:
You can use `tqdm` here if you want:

```suggestion
    for item in tqdm.tqdm(scan_table(dynamodb_client, TableName=table_name), total=total_bags):
        dynamo_id = item['id']['S']
        version = int(item['version']['N'])
        stored_version = bags.get(dynamo_id, -1)

        if version > stored_version:
            bags[dynamo_id] = version
```